### DISCUSSION TITLE
Slow response times in dev

### DISCUSSION THREAD

**Discussion by binajmen** [remix-run/remix#3335] [Q&A]
Hi 👋 

I've been off for a week. This morning, I ran my project in dev mode (`npm run dev`) and saw massive response times: 1500ms at best..

I've made close to no (big) changes today, so I'm not sure what happened. Strangely enough, I've pushed the project as it is to fly and the response times are (seem?) as usual: lightning fast ⚡.

My laptop was running the project fine one week ago, and current CPU usage is normal. I have no clue what is going on. 

I digged in my git log and located the difference between two commits.

On my commit **470c48d**:
```
HEAD / 200 - - 106.042 ms
GET /healthcheck 200 - - 219.782 ms
HEAD / 200 - - 148.995 ms
GET /healthcheck 200 - - 256.889 ms
HEAD / 200 - - 105.246 ms
GET /healthcheck 200 - - 213.265 ms
```

On the following commit **f8235e6**:
```
HEAD / 200 - - 1459.199 ms
GET /healthcheck 200 - - 3409.599 ms
HEAD / 200 - - 1145.845 ms
GET /healthcheck 200 - - 2367.672 ms
HEAD / 200 - - 1353.397 ms
GET /healthcheck 200 - - 3228.087 ms
```

> FYI, the `/healthcheck` is the one from the Blues stack.

If I perform a `git diff` between the two commits, I noticed I've installed three new libraries:
```
+        "cloudinary": "^1.30.0",
+        "jsdom": "^19.0.0",
+        "qrcode": "^1.5.0",
```

How can I profile if one of this library is the cause? I tried to use the DevTools for Node.js Profiler from Chrome, but I'm unable to interpret the content, although it is clear way much more is happening:

For the commit **470c48d**:
<img width="944" alt="Screenshot 2022-05-30 at 14 31 37" src="https://user-images.githubusercontent.com/15611419/170992946-f19b7657-9402-4532-8d66-c4e9939f5f46.png">

For the commit **f8235e6**:
<img width="964" alt="Screenshot 2022-05-30 at 14 29 28" src="https://user-images.githubusercontent.com/15611419/170993170-eda679a1-8bc0-44c4-98c7-5471f65df3e9.png">

Noticed the `transpileModule` thing happenng? Could it be that a module needs heavy work in dev mode?

Any help would be greatly appreciated 🙏 🙏

**Comment by binajmen** [comment]
So apparently running `npm run dev` with my commit `470c48d` lead to the following message:
```
The following suggested values were added to your "tsconfig.json". These values can be changed to fit your project's needs:

        - compilerOptions.allowJs was set to 'true'
        - compilerOptions.forceConsistentCasingInFileNames was set to 'true'
```

If I switch `allowJs` to `false`, the response time come back to normal.

My questions would be:
1. Why these suggested values were added?
2. What is the impact of `allowJs`?
3. How can I identify which module/library is influenced by this option?
4. Bonus question: why was it not slow down one week ago? 🤨

**Comment by nielskrijger** [comment]
My response times were even worse, over 2 seconds on a Macbook Pro 2019. The app is very small so I'm not sure what's going on.

Thank you for the workaround, it significantly speeds up the build!

This should be addressed, the dev experience of the Blues stack https://github.com/remix-run/blues-stack became very frustrating after a good week of developing in it, to the point I was going to give up and switch frameworks until I found this thread.
